src/solver/variable/state.cpp

```diff
@@ -353,117 +352,149 @@ void State::yearEndBuildFromThermalClusterIndex(const uint clusterAreaWideIndex,
     }
 
     if (dur > 0)
+        yearEndBuildFromThermalClusterIndexDurPositive(dur, ON_min, ON_max, ON_opt);
+
+    // Calculation of non linear and startup costs
+    yearEndBuildThermalClusterCalculateStartupCosts(dur, ON_min, ON_opt, currentCluster);
+}
+
+void State::yearEndBuildThermalClusterCalculateStartupCosts(uint& dur,
+                std::array<uint, Variable::maxHoursInAYear>& ON_min,
+                std::array<uint, Variable::maxHoursInAYear>& ON_opt,
+                Data::ThermalCluster* currentCluster)
+{
+    uint startHourForCurrentYear = study.runtime->rangeLimits.hour[Data::rangeBegin];
+    uint endHourForCurrentYear
+        = startHourForCurrentYear + study.runtime->rangeLimits.hour[Data::rangeCount];
+
+    for (uint i = startHourForCurrentYear; i < endHourForCurrentYear; ++i)
     {
-        ON_opt[startHourForCurrentYear] = ON_min[startHourForCurrentYear];
-        h = startHourForCurrentYear + 1;
+        double thermalClusterStartupCostForYear = 0;
+        double thermalClusterFixedCostForYear = 0;
 
-        while (h < endHourForCurrentYear)
+        uint optimalCount;
```

Comment:
Instead of : 
```cpp
uint optimalCount;
(dur == 0) ? (optimalCount = ON_min[i]) : (optimalCount = ON_opt[i]);
```
we should have : 
```cpp
uint optimalCount = (! dur) ? ON_min[i] : ON_opt[i];
```
And we should replace **i** with **hour**.